adding profile_ivalue

jjsjann123 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47668 profile ivalue for nvfuser
* #47667 refactor profiling optional
* **#47666 adding profile_ivalue**

Differential Revision: [D25255573](https://our.internmc.facebook.com/intern/diff/D25255573)

jjsjann123 committed:
adding profile_ivalue

[ghstack-poisoned]

eellison reviewed (commented):
Looks fine to me, wait on @Krovatkin

jjsjann123 commented on torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp:
```diff
@@ -36,6 +36,16 @@ RegisterOperators reg(
+     Operator(
+         prim::profile_ivalue,
+         [](const Node* node) -> Operation {
+           auto callback = node->cast<ProfileIValueOp>()->getCallback();
```
Good catch. It probably should be removed now since profile is lowered to interpreter's instruction. Same with prim::profile & prim::profile_optional :stuck_out_tongue:

jjsjann123 committed:
Update on "adding profile_ivalue"




[ghstack-poisoned]

eellison reviewed (approved):
LGTM, some tests never hurt but i guess this is used in the next PR

jjsjann123 commented:
@Krovatkin 
I know we are still going through the refactor PR. Meanwhile, this one should be good to go. Can we try to import it?

jjsjann123 commented:
Having said that, if there's any real failure marked from the merge conflict in CI, feel free to point it to me so I can try to fix.

jjsjann123 committed:
Update on "adding profile_ivalue"


Differential Revision: [D25255573](https://our.internmc.facebook.com/intern/diff/D25255573)

[ghstack-poisoned]

jjsjann123 committed:
Update on "adding profile_ivalue"


Differential Revision: [D25255573](https://our.internmc.facebook.com/intern/diff/D25255573)

[ghstack-poisoned]

jjsjann123 committed:
Update on "adding profile_ivalue"


Differential Revision: [D25255573](https://our.internmc.facebook.com/intern/diff/D25255573)

[ghstack-poisoned]
